src/adapter/src/coord/sequencer.rs

```diff
@@ -3989,6 +3995,25 @@ impl Coordinator {
         }
         Ok(ops)
     }
+
+    /// Installs the in-memory state for the cluster linked to the specified
+    /// object after a create operation, if such a linked cluster exists.
+    pub(crate) async fn maybe_create_linked_cluster(&mut self, linked_object_id: GlobalId) {
+        if let Some(cluster) = self.catalog.get_linked_cluster(linked_object_id) {
+            let name = cluster.name().to_string();
+            self.create_compute_instance(&name).await;
+        }
+    }
+
+    /// Updates the in-memory state for the replicas of the cluster linked to
+    /// the specified object after an alter operation, if such a linked cluster
+    /// exists.
+    pub(crate) async fn maybe_alter_linked_cluster(&mut self, linked_object_id: GlobalId) {
+        if let Some(cluster) = self.catalog.get_linked_cluster(linked_object_id) {
+            let cluster_name = cluster.name().to_string();
+            self.create_compute_replica(&cluster_name, LINKED_CLUSTER_REPLICA_NAME).await;
```

Comment:
I'm a little confused by this. Why do we create a new replica when altering the existing cluster?

Reply:
Added a comment explaining the situation!